interfaces/cython/cantera/examples/reactors/reactor1.py

```diff
@@ -12,16 +12,26 @@
 r = ct.IdealGasConstPressureReactor(gas)
 
 sim = ct.ReactorNet([r])
+sim.initialize()
```

Comment:
I'm not keen on requiring users to explicitly call the `initialize` function. Is there a reason not to simply have `setAdvanceLimit` call `initialize` if needed, as we do in other methods such as `ReactorNet::advance` and `ReactorNet::globalComponentIndex`?

Reply:
Good point. I opted for this as the `setAdvanceLimit` is called on a `Reactor` object, not `ReactorNet`. However, I found a way to eliminate this call by using a callback from `Reactor` via `Reactor:m_net`, which allows for the required `ReactorNet::initialize` call.